src/CodeFormatter/FormatOptions.cs

```diff
@@ -117,21 +108,24 @@ private static ImmutableArray<string> InitializeCopyrightHeaderText(string copyr
         private ImmutableDictionary<string, bool> BuildRuleMapFromOptions()
         {
             _ruleMap = ImmutableDictionary<string, bool>.Empty;
-            if (EnabledRules != null)
+
```

Comment:
Rather than using an `ImmutableDictionary` (which you then have to recreate each time you call `SetItem` on it, how about using a plain `Dictionary` and then making an `ImmutableDictionary` from it at the end?